src/UserService.Business/Commands/Image/AddImagesCommand.cs

```diff
@@ -0,0 +1,180 @@
+using LT.DigitalOffice.Models.Broker.Enums;
+using LT.DigitalOffice.Models.Broker.Requests.Image;
+using LT.DigitalOffice.Models.Broker.Responses.Image;
+using LT.DigitalOffice.UserService.Business.Commands.Image.Interfaces;
+using LT.DigitalOffice.UserService.Data.Interfaces;
+using LT.DigitalOffice.UserService.Mappers.Db.Interfaces;
+using LT.DigitalOffice.UserService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.UserService.Models.Db;
+using LT.DigitalOffice.UserService.Models.Dto.Enums;
+using LT.DigitalOffice.UserService.Models.Dto.Requests.User;
+using LT.DigitalOffice.UserService.Models.Dto.Requests.User.Images;
+using LT.DigitalOffice.UserService.Validation.Image.Interfaces;
+using MassTransit;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.UserService.Business.Commands.Image
+{
+  public class AddImagesCommand : IAddImagesCommand
+  {
+    private readonly IImageRepository _imageRepository;
+    private readonly ICertificateRepository _certificateRepository;
+    private readonly IEducationRepository _educationRepository;
+    private readonly IAccessValidator _accessValidator;
+    private readonly IAddImagesRequestValidator _requestValidator;
+    private readonly IDbEntityImageMapper _dbEntityImageMapper;
+    private readonly ICreateImageDataMapper _createImageDataMapper;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly IRequestClient<ICreateImagesRequest> _rcCreateImage;
+    private readonly ILogger<AddImagesCommand> _logger;
+
+    private async Task<List<Guid>> AddImages(List<AddImageRequest> request, Guid senderId, List<string> errors)
+    {
+      if (request == null || !request.Any())
+      {
+        return null;
+      }
+
+      const string errorMessage = "Can not add images. Please try again later.";
+      const string logMessage = "Errors while adding images.";
+
+      try
+      {
+        Response<IOperationResult<ICreateImagesResponse>> createResponse = await _rcCreateImage.GetResponse<IOperationResult<ICreateImagesResponse>>(
+          ICreateImagesRequest.CreateObj(
+            _createImageDataMapper.Map(request, senderId),
+            ImageSource.User), default, TimeSpan.FromSeconds(5));
+
+        if (createResponse.Message.IsSuccess)
+        {
+          return createResponse.Message.Body.ImagesIds;
+        }
+
+        _logger.LogWarning(
+          logMessage + " Errors: {Errors}",
+          string.Join('\n', createResponse.Message.Errors));
+
+        errors.Add(errorMessage);
+      }
+      catch (Exception e)
+      {
+        _logger.LogError(e, logMessage);
+
+        errors.Add(errorMessage);
+      }
+
+      return null;
+    }
+
+    private Guid? GetUserIdFromEntity(Guid entityId, EntityType entityType)
+    {
+      Guid? userId = null;
+
+      switch (entityType)
```

Comment:
you can do it with switch expression 
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression
i guess it will be better

Reply:
yes, it is better, thanks